### PR TITLE
Add e2e test for on-error subgraph

### PR DESCRIPTION
**Pull request by gluke77**
This change is [Reviewable](https://reviewable.io/reviews/mirantis/k8s-appcontroller/196)

### CODE REVIEW

File: e2e/utils/utils.go [hunk 1]
```diff
@@ -82,10 +88,20 @@ func WaitForPod(clientset *kubernetes.Clientset, namespace string, name string,
...
 
+func WaitForPodNotToBeCreated(clientset *kubernetes.Clientset, namespace string, name string) {
+	defer GinkgoRecover()
+	Consistently(func() int32 {
+		_, err := clientset.Core().Pods(namespace).Get(name)
+		statusError, ok := err.(*errors.StatusError)
+		Expect(ok).To(BeTrue())
+		return statusError.ErrStatus.Code
+	}, Duration, PollingInterval).Should(BeNumerically("==", 404))
+}
```
Comment (dshulyak) [inline e2e/utils/utils.go]: This is way too long. 5 seconds should be enough to guarantee that pod is not created

File: e2e/utils/utils.go [hunk 2]
```diff
@@ -82,10 +88,20 @@ func WaitForPod(clientset *kubernetes.Clientset, namespace string, name string,
...
 		return nil
-	}, 120*time.Second, 5*time.Second).Should(BeNil())
+	}, Duration, PollingInterval).Should(BeNil())
 	return podUpdated
 }
 
+func WaitForPodNotToBeCreated(clientset *kubernetes.Clientset, namespace string, name string) {
+	defer GinkgoRecover()
+	Consistently(func() int32 {
+		_, err := clientset.Core().Pods(namespace).Get(name)
```
Comment (dshulyak) [inline e2e/utils/utils.go]: it is better to check with IsNotFound from api/errors package

File: e2e/basic_test.go [hunk 3]
```diff
@@ -269,3 +306,7 @@ func PodPause(name string) *v1.Pod {
 		},
 	}
 }
+
+func FailedPod(name string) *v1.Pod {
```
Comment (dshulyak) [inline e2e/basic_test.go]: it is not used anywhere

**Comment by gluke77** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[e2e/utils/utils.go, line 98 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/196#-KbzcvkXKnEwfSoZSaHQ-r1-98:-Kc2ho3SD07oGRKJeTLg:b-9n66ir) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/89caeb993a9fa58a3096e4918b2ba1547cd24749/e2e/utils/utils.go#L98)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

better to use IsNotFound from client-go/api/errors package (pkg name might be not accurate ;))
</blockquote></details>

Awesome, that's what I was looking for

---

*[e2e/utils/utils.go, line 103 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/196#-KbzcvkXKnEwfSoZSaHQ-r1-103:-Kc2hv6yq1mTYvVjQJ_b:bqewkut) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/89caeb993a9fa58a3096e4918b2ba1547cd24749/e2e/utils/utils.go#L103)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

This is way too long. 5 seconds should be enough to guarantee that pod is not created
</blockquote></details>

sounds reasonable

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/196)*

**Comment by gluke77** [comment]
Review status: 0 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[e2e/basic_test.go, line 310 at r1](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/196#-KbzcvkXKnEwfSoZSaHP-r1-310:-Kc2mAcjvGNvKsVWmOVq:bb7f2mf) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/89caeb993a9fa58a3096e4918b2ba1547cd24749/e2e/basic_test.go#L310)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

it is not used anywhere
</blockquote></details>

done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/196)*

**Comment by dshulyak** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/196)*

**Comment by dshulyak** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/196#-:-Kc2pYbErJ1kTHebVWHp:bnfp4nl)*

**Comment by pigmej** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/196#-:-Kc2qoIKNYe-savhBnlS:bnfp4nl)*

**Pull request merged**